Builds/containers/ubuntu-builder/Dockerfile

```diff
@@ -0,0 +1,33 @@
+ARG DIST_TAG=16.04
+ARG GIT_COMMIT=unknown
+
+FROM ubuntu:$DIST_TAG
+LABEL git-commit=$GIT_COMMIT
+
+# install/setup prerequisites:
+COPY ubuntu-builder/ubuntu_setup.sh /tmp/
+COPY shared/build_deps.sh /tmp/
+COPY shared/install_cmake.sh /tmp/
+RUN chmod +x /tmp/ubuntu_setup.sh
+RUN chmod +x /tmp/build_deps.sh
+RUN chmod +x /tmp/install_cmake.sh
+RUN /tmp/ubuntu_setup.sh
+RUN /tmp/install_cmake.sh
```

Comment:
Nit: I'd link the run's with `&&`

Reply:
I agree the `chmod`s can be chained...but the various setup scripts do quite a bit of work and I was hoping to take advantage of intermediate layers (in the case of one step failing...). That might not be a sound strategy, so happy to revisit/optimize this later.